### PR TITLE
Apps/mavencentral publish

### PR DESCRIPTION
**Pull request by manolovn**
### :goal_net: What's the goal?
Provide a new github action to publish library compose version and catalogo compose version to maven central repositories.

### ☑️ Checks
- [ ] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
- [x] Action tested manually

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
         required: true
 jobs:
   release:
-    runs-on: self-hosted-novum
+    runs-on: [self-hosted-org, linux]
```
Comment (dpastor) [inline .github/workflows/release.yml:13]: Suggested by Lifecycle?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -33,6 +33,12 @@ jobs:
         with:
           args: 'bash ./gradlew clean :library:assembleRelease :catalog:assembleRelease'
 
+      - name: Build Compose library
```
Comment (dpastor) [inline .github/workflows/release.yml:36]: Can't be done on previous step? so it is faster

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -45,6 +51,18 @@ jobs:
         with:
           args: 'bash ./gradlew :library:publishReleasePublicationToSonatypeRepository -DLIBRARY_VERSION=${{ github.event.inputs.libraryVersion }}'
 
+      - name: Release Compose library
```
Comment (dpastor) [inline .github/workflows/release.yml:54]: Could be done in previous step also no?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -57,6 +75,18 @@ jobs:
         with:
           args: 'bash ./gradlew :catalog:publishCatalogPublicationToSonatypeRepository -DLIBRARY_VERSION=${{ github.event.inputs.libraryVersion }}'
 
+      - name: Release Compose catalog
```
Comment (dpastor) [inline .github/workflows/release.yml:78]: Same thing 😄

File: mavencentral.gradle [hunk 5]
```diff
@@ -73,6 +115,56 @@ publishing {
...
+                scm {
+                    connection = 'scm:git:github.com/Telefonica/mistica-android.git'
+                    developerConnection = 'scm:git:ssh://github.com/Telefonica/mistica-android.git'
+                    url = 'https://github.com/Telefonica/mistica-android/tree/master'
+                }
+                withXml {
+                    def dependenciesNode = asNode().appendNode('dependencies')
+
+                    project.configurations.getByName("implementation").allDependencies.each {
+                        if (it.group != "com.telefonica.mistica" && it.name != "library") {
```
Comment (dpastor) [inline mavencentral.gradle:144]: Shouldn't this be library-compose? (In the catalogCompose section, as this has calculated the diff adding new code here 😄)

Comment (gmerinojimenez) [reply mavencentral.gradle:144]: I think we should exclude both

File: mavencentral.gradle [hunk 6]
```diff
@@ -65,6 +65,48 @@ publishing {
...
+                }
+                scm {
+                    connection = 'scm:git:github.com/Telefonica/mistica-android.git'
+                    developerConnection = 'scm:git:ssh://github.com/Telefonica/mistica-android.git'
+                    url = 'https://github.com/Telefonica/mistica-android/tree/master'
+                }
+                withXml {
+                    def dependenciesNode = asNode().appendNode('dependencies')
+
+                    project.configurations.getByName("implementation").allDependencies.each {
```
Comment (gmerinojimenez) [inline mavencentral.gradle:101]: make sure that we're adding "library" correctly here

File: .github/workflows/release.yml (hunk 1 above)
Comment (manolovn) [reply .github/workflows/release.yml]: reverted finally

Comment (manolovn) [reply .github/workflows/release.yml]: nope, finally reverted

File: mavencentral.gradle (hunk 5 above)
Comment (manolovn) [reply mavencentral.gradle]: I've excluded both

File: .github/workflows/release.yml (hunk 4 above)
Comment (manolovn) [reply .github/workflows/release.yml]: done

File: .github/workflows/release.yml (hunk 3 above)
Comment (manolovn) [reply .github/workflows/release.yml]: done

File: .github/workflows/release.yml (hunk 2 above)
Comment (manolovn) [reply .github/workflows/release.yml]: done

**Review by dpastor — APPROVED** [review]

**Review by gmerinojimenez — APPROVED** [review]
Nice!!!

**Comment by manolovn** [comment]
manually tested and it published to staging correctly
image

**Pull request merged**